### PR TITLE
Add a fix for gradient aggregation

### LINKED ISSUE #51: 🐛[BUG]: Gradient Aggregation does not work as expected
### Version

1.1.0

### On which installation method(s) does this occur?

Docker, Pip, Source

### Describe the issue

The gradient aggregation feature does not seem to improve the results over the baseline results, indicating some bug. 

Orange: Baseline 

Green: 0.25x Batchsize

Grey: 0.25x Batchsize + 4 Gradient aggregations.

image

One would expect the results of the Grey case to match with the Orange case, but that does not seem to be the case currently. 

### Minimum reproducible example

```shell
This was tested on annular ring example
```

### Relevant log output

_No response_

### Environment details

_No response_

### Other/Misc.

_No response_

### PR DESCRIPTION
**Pull request by ktangsali**
# Modulus Pull Request

## Description
The Gradient aggregation was not functioning correctly, because it was computing the losses on the same batch as opposed to different batches that is needed for Gradient aggregation. This PR adds a fix that enables use of Gradient aggregation for cases without CUDA Graphs. 

Closes #51 

Annular ring case now works as expected with Gradient aggregation. 

Pink: Baseline, Blue: 0.1x batch size, Red: 0.1x batch size + 10 gradient aggregation steps.
 
image

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus-sym/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus-sym/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus-sym/issues) is linked to this pull request.

### CODE REVIEW

File: modulus/sym/trainer.py [hunk 1]
```diff
@@ -60,11 +60,21 @@ def adam_compute_gradients(
         self, aggregator: nn.Module, global_optimizer_model: nn.Module, step: int
     ):
         loss, losses = 0, Counter({})
+
+        if self.cfg.cuda_graphs and self.grad_agg_freq != 1:
+            self.log.warn(
```
Comment (mnabian) [inline modulus/sym/trainer.py:65]: Can we please make an issue to track this?
Also, I prefer raising an exception instead of a warning and forcing `grad_agg_freq` to 1.

File: modulus/sym/trainer.py [hunk 2]
```diff
@@ -60,11 +60,21 @@ def adam_compute_gradients(
...
+                "supported currently. Setting grad_agg_freq = 1."
+            )
+            self.grad_agg_freq = 1
+
         for agg_step in range(self.grad_agg_freq):
             with torch.autocast(
                 self.device_amp, enabled=self.amp, dtype=self.amp_dtype
             ):
                 torch.cuda.nvtx.range_push("Loss computation")
+                if agg_step != 0:  # load new data for subsequent steps
```
Comment (mnabian) [inline modulus/sym/trainer.py:76]: I suggest we take this outside of the nvtx range for loss computation.

**Review by mnabian — APPROVED** [review]

File: modulus/sym/trainer.py (hunk 1 above)
Comment (ktangsali) [reply modulus/sym/trainer.py]: Changed it to a `ValueError`.

File: modulus/sym/trainer.py (hunk 2 above)
Comment (ktangsali) [reply modulus/sym/trainer.py:73]: Thanks, done
